examples/nested-folders/src/Folder.js

```diff
@@ -78,29 +78,19 @@ function view(state$, childrenVDOM$) {
     })
 }
 
-function Children(sources) {
-  const array$ = sources.onion.state$
-
-  const childrenSinks$ = array$.map(array =>
-    array.map((item, index) => isolate(Folder, index)(sources))
-  )
-
-  const childrenReducer$ = childrenSinks$
-    .compose(pick('onion'))
-    .compose(mix(xs.merge))
-
-  const childrenVDOM$ = childrenSinks$
-    .compose(pick('DOM'))
-    .compose(mix(xs.combine))
-
-  return {
-    DOM: childrenVDOM$,
-    onion: childrenReducer$,
-  }
-}
-
 export default function Folder(sources) {
+  const Children = makeCollection({
```

Comment:
I would leave `Children` outside of Folder

Reply:
@jvanbruegge I wrote it like this:
~~~
function Children(sources = {}) {
  const children = makeCollection({
    item: Folder,
    itemKey: state => state.id,
    itemScope: key => key,
    collectSinks: instances => ({
      onion: instances.pickMerge('onion'),
      DOM: instances.pickCombine('DOM')
    })
  })

  return children
}

export default function Folder(sources) {
  const childrenSinks = isolate(Children(), 'children')(sources)
  ...
~~~
Then had kind of mixed feelings since I just started learning this awesome framework. So I thought it would make sense to move the code inside the `Folder` to highlight the implicit creation of the `Children` component.